client/network_connect.go

```diff
@@ -6,23 +6,35 @@ import (
 	"github.com/moby/moby/api/types/network"
 )
 
+// NetworkConnectOptions represents the data to be used to connect a container to the
+// network.
+type NetworkConnectOptions struct {
+	Container      string
+	EndpointConfig *network.EndpointSettings `json:",omitempty"`
+}
+
+// NetworkConnectResult represents the result of a NetworkConnect operation.
+type NetworkConnectResult struct {
+	// Currently empty; placeholder for future fields.
+}
+
 // NetworkConnect connects a container to an existent network in the docker host.
-func (cli *Client) NetworkConnect(ctx context.Context, networkID, containerID string, config *network.EndpointSettings) error {
+func (cli *Client) NetworkConnect(ctx context.Context, networkID, containerID string, options NetworkConnectOptions) (NetworkConnectResult, error) {
 	networkID, err := trimID("network", networkID)
 	if err != nil {
-		return err
+		return NetworkConnectResult{}, err
 	}
 
 	containerID, err = trimID("container", containerID)
 	if err != nil {
-		return err
+		return NetworkConnectResult{}, err
 	}
 
 	nc := NetworkConnectOptions{
 		Container:      containerID,
-		EndpointConfig: config,
+		EndpointConfig: options.EndpointConfig,
```

Comment:
@thaJeztah , I think this might be another case where I mistook the name on the type and moved a request out of API. WDYT?